compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt

```diff
@@ -277,34 +299,174 @@ private fun Modifier.pointerScrollable(
         },
         canDrag = { down -> down.type != PointerType.Mouse }
-    )
-        .mouseWheelScroll(scrollLogic, scrollConfig)
-        .nestedScroll(nestedScrollConnection, nestedScrollDispatcher.value)
+    ).run {
+        if (scrollConfig.isSmoothScrollingEnabled) {
+            animatedMouseWheelScroll(scrollLogic, scrollConfig)
+        } else {
+            rawMouseWheelScroll(scrollLogic, scrollConfig)
+        }
+    }.nestedScroll(nestedScrollConnection, nestedScrollDispatcher.value)
 }
 
-private fun Modifier.mouseWheelScroll(
-    scrollingLogicState: State<ScrollingLogic>,
+@Composable
+private fun Modifier.rawMouseWheelScroll(
+    scrollLogic: State<ScrollingLogic>,
     mouseWheelScrollConfig: ScrollConfig,
-) = pointerInput(scrollingLogicState, mouseWheelScrollConfig) {
-    awaitPointerEventScope {
-        while (true) {
-            val event = awaitScrollEvent()
-            if (event.changes.fastAll { !it.isConsumed }) {
-                with(mouseWheelScrollConfig) {
-                    val scrollAmount = calculateMouseWheelScroll(event, size)
-                    with(scrollingLogicState.value) {
-                        val delta = scrollAmount.toFloat().reverseIfNeeded()
-                        val consumedDelta = scrollableState.dispatchRawDelta(delta)
-                        if (consumedDelta != 0f) {
-                            event.changes.fastForEach { it.consume() }
-                        }
+) = mouseWheelInput(scrollLogic, mouseWheelScrollConfig) { event ->
+    val delta = with(mouseWheelScrollConfig) {
+        calculateMouseWheelScroll(event, size)
+    }
+    scrollLogic.value.dispatchRawDelta(delta) != Offset.Zero
+}
+
+@Composable
+private fun Modifier.animatedMouseWheelScroll(
+    scrollLogic: State<ScrollingLogic>,
+    mouseWheelScrollConfig: ScrollConfig,
+): Modifier {
+    val density = LocalDensity.current.density
+    var isAnimationRunning by remember { mutableStateOf(false) }
+    val channel = remember { Channel<Float>(capacity = Channel.UNLIMITED) }
+    LaunchedEffect(scrollLogic) {
+        while (isActive) {
+            val event = channel.receive()
+            isAnimationRunning = true
+            scrollLogic.value.animatedDispatchScroll(event, speed = 1f * density) {
+                // Sum delta from all pending events to avoid multiple animation restarts.
+                channel.sumOrNull()
+            }
+            isAnimationRunning = false
+        }
+    }
+    return mouseWheelInput(scrollLogic, mouseWheelScrollConfig) { event ->
+        val scrollDelta = with(mouseWheelScrollConfig) {
+            calculateMouseWheelScroll(event, size)
+        }
+        if (mouseWheelScrollConfig.isPreciseWheelScroll(event)) {
+            // In case of high-resolution wheel, such as a freely rotating wheel with no notches
+            // or trackpads, delta should apply directly without any delays.
+            scrollLogic.value.dispatchRawDelta(scrollDelta) != Offset.Zero
```

Comment:
Because we changed dispatchRawDelta to scroll for a non-precise mouse, shouldn't we change that for a precise mouse as well? So, nestedScrollable always works, not only when there is an animation.

Reply:
It already do. nested scroll works in both cases via not consuming event